cmake/Options.cmake

```diff
@@ -52,11 +52,32 @@ function(set_options)
         set(DEFAULT_OPTION_INSTALL_FHS OFF)
     endif()
 
+    # Turn off asserts only for official releases
+    # Quick and dirty method of determining whether or not this is a tagged (stable) release
+    # Structure largely copied from FindVersion.cmake
+    set(DEFAULT_OPTION_USE_ASSERTS ON)
+    find_package(Git QUIET)
+    if(GIT_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
+        # Get the tag
+        execute_process(COMMAND ${GIT_EXECUTABLE} name-rev --name-only --tags --no-undefined HEAD
+                        OUTPUT_VARIABLE TAG
+                        OUTPUT_STRIP_TRAILING_WHITESPACE
+                        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+                        ERROR_QUIET
+        )
+        if(NOT ${TAG} STREQUAL "")
+            string(REGEX REPLACE "^[0-9.]+$" "" STABLETAG "${TAG}")
+            if(STABLETAG STREQUAL "")
+                set(DEFAULT_OPTION_USE_ASSERTS OFF)
+            endif()
+        endif()
+    endif()
```

Comment:
I like the approach, I very much dislike the location. Should be outside this function for sure (which is fine in CMake, you just need to do this before this function is called).